### PR TITLE
Documentation: fix typo, add to features list

### PR DESCRIPTION
**Pull request by tooomm**
## Proposed change

https://github.com/paperless-ngx/paperless-ngx/pull/4542 reworked the list of features and moved it to the docs.

I made some additions:
- Stress that files are searchable after OCR
- Highlight amount of supported languages and engine used
- Mention PDF/A for archive storage and that originals are not changed

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other (please explain): Docs

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [ ] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [ ] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [ ] I have checked my modifications for any breaking changes.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -18,7 +18,9 @@ physical documents into a searchable online archive so you can keep, well, _less
 ## Features
 
 -   **Organize and index** your scanned documents with tags, correspondents, types, and more.
--   Performs **OCR** on your documents, adding selectable text to image-only documents.
+-   Performs **OCR** on your documents, adding selectable text to image-only documents and makes their content searchable.
+    - Paperless utilizes the open-source Tesseract engine to recognize more than 100 languages.
```
Comment (shamoon) [inline docs/index.md:22]:
```suggestion
-   Utilizes the open-source Tesseract engine to recognize more than 100 languages.
```

File: docs/index.md [hunk 2]
```diff
@@ -18,7 +18,9 @@ physical documents into a searchable online archive so you can keep, well, _less
...
+-   All documents are saved as PDF/A, a special file format intended for long term storage, alongside the unaltered originals.
```
Comment (shamoon) [inline docs/index.md:23]:
```suggestion
-   Documents are saved as PDF/A format which is designed for long term storage, alongside the unaltered originals.
```

File: docs/index.md [hunk 3]
```diff
@@ -18,7 +18,9 @@ physical documents into a searchable online archive so you can keep, well, _less
 ## Features
 
 -   **Organize and index** your scanned documents with tags, correspondents, types, and more.
--   Performs **OCR** on your documents, adding selectable text to image-only documents.
+-   Performs **OCR** on your documents, adding selectable text to image-only documents and makes their content searchable.
```
Comment (shamoon) [inline docs/index.md:21]:
```suggestion
-   Performs **OCR** on your documents, adding searchable and selectable text, to even documents scanned with only images.
```

**Review by shamoon — CHANGES_REQUESTED** [review]

**Review by shamoon — APPROVED** [review]

**Comment by tooomm** [comment]
Oh, I see there was a spacing issue after the bullet point.

👍 for your slight wording adjustments, short and precise.
Thank you, shamoon!

**Pull request merged**